torch/_dynamo/decorators.py

```diff
@@ -155,8 +156,16 @@ def allow_in_graph(fn):
         return [allow_in_graph(x) for x in fn]
     assert callable(fn), "allow_in_graph expects a callable"
     if trace_rules.lookup_callable(fn) != variables.TorchInGraphFunctionVariable:
-        trace_rules._disallowed_callable_ids.remove(id(fn))
-        trace_rules._allowed_callable_ids.add(id(fn))
+        fn_id = id(fn)
+        trace_rules._disallowed_callable_ids.remove(fn_id)
+        trace_rules._allowed_callable_ids.add(fn_id)
+
+        # Avoid id reuse which creates subtle bugs.
+        def deregister():
+            trace_rules._allowed_callable_ids.remove(fn_id)
+            trace_rules._nonstrict_trace_callable_ids.remove(fn_id)
```

Comment:
do we need to remove stuff from _disallowed_callable_ids?

Reply:
OOps copy paste error, updating